src/ops/shell.rs

```diff
@@ -14,17 +14,40 @@ use std::process::Command;
 use std::time::{Duration, Instant};
 use std::{env, fs, thread};
 
-/// See the documentation for lorri::cli::Command::Shell for more
-/// details.
+/// This is the entry point for the `lorri shell` command.
+///
+/// # Overview
+///
+/// `lorri shell` launches the user's shell with the project environment set up. "The user's shell"
+/// here just means whatever binary $SHELL points to. Concretely this involves the following:
+///
+/// 1. `lorri shell` builds the project environment and writes a bash init script that loads the
+///    project environment
+/// 2. `lorri shell` spawn a bash process with the init script as its `--rcfile`
+/// 3. bash execs into `lorri start_user_shell_`
+/// 4. `lorri start_user_shell_` performs shell-specific setup for $SHELL
+/// 5. `lorri start_user_shell_` execs into the user shell $SHELL
+///
+/// This setup allows lorri to support almost any shell with minimal additional work. Only step 4
+/// must be adjusted, and only in case we want to customize the shell, e.g. changing the way the
+/// prompt looks.
 pub fn main(project: Project) -> OpResult {
+    let lorri = env::current_exe().expect("failed to determine lorri executable's path");
     let shell = env::var("SHELL").expect("lorri shell requires $SHELL to be set");
     debug!("using shell path {}", shell);
 
     let tempdir = tempfile::tempdir().expect("failed to create temporary directory");
     let mut bash_cmd = bash_cmd(project, tempdir.path())?;
     debug!("bash"; "command" => ?bash_cmd);
     bash_cmd
-        .args(&["-c", &format!("exec {}", shell)])
+        .args(&[
+            "-c",
+            &format!(
+                "exec {} start_user_shell_ --shell-path=\"{}\"",
```

Comment:
```
$ bash -c 'echo "$1" "$2"' -- foo bar
foo bar
```

Replace both `{}`s with "$1" and "$2" then add three more args to the program: `--`, `lorri.display()`, `shell`

Reply:
Done, thanks!